src/app/resources/resources-add.component.ts

```diff
@@ -0,0 +1,196 @@
+    private resourceValidatorService: ResourceValidatorService,
+    private userService: UserService
+  ) {
+    this.createForm();
+  }
+
+  createForm() {
+    this.resourceForm = this.fb.group({
+      title: [
+        '',
+        Validators.required,
+        // an arrow function is for lexically binding 'this' otherwise 'this' would be undefined
+        ac => this.resourceValidatorService.checkResourceExists$(ac)
+      ],
+      author: '',
+      year: '',
+      language: '',
+      publisher: '',
+      linkToLicense: '',
+      subject: [ '', Validators.required ],
+      level: [ '', Validators.required ],
+      openWith: '',
+      resourceFor: [],
+      medium: '',
+      articleDate: '',
+      resourceType: '',
+      addedBy: '',
+      openUrl: [],
+      openWhichFile: ''
+    });
+  }
+
+  onSubmit() {
+    if (this.resourceForm.valid) {
+      if (this.file !== undefined) {
+        if ( this.file.type.indexOf('image') > -1 ) {
+          this.mediaType = 'image';
+        } else if (this.file.type.indexOf('pdf') > -1) {
+          this.mediaType = 'pdf';
+        } else if (this.file.type.indexOf('audio') > -1) {
+          this.mediaType = 'audio';
+        } else if (this.file.type.indexOf('video') > -1) {
+          this.mediaType = 'video';
+        } else if (this.file.type.indexOf('zip') > -1) {
+          this.zipAttachment();
+        } else {
+          this.mediaType = 'other';
+        }
+        if (this.file.type.indexOf('zip') === -1 ) {
+          const reader = new FileReader(),
+          rComp = this;
+          reader.readAsDataURL(this.file);
+          reader.onload = () => {
+          // FileReader result has file type at start of string, need to remove for CouchDB
+          const fileData = reader.result.split(',')[1],
+          attachments = {};
+          attachments[rComp.file.name] = {
+            content_type: rComp.file.type,
+            data: fileData
+          };
+          const resource = Object.assign({ },
+            {
+              filename: rComp.file.name,
+              _attachments: attachments,
+              mediaType: rComp.mediaType
+            }
+          );
+          this.addResource(Object.assign(this.resourceForm.value, resource));
+          };
+        }
+      } else {
+        this.addResource(this.resourceForm.value);
+      }
+    } else {
+      Object.keys(this.resourceForm.controls).forEach(field => {
+        const control = this.resourceForm.get(field);
+        control.markAsTouched({ onlySelf: true });
+      });
+    }
+  }
+
+  async addResource(resourceInfo) {
+    // ...is the rest syntax for object destructuring
+    try {
+      await this.couchService.post(this.dbName, { ...resourceInfo });
+        this.router.navigate([ '/resources' ]);
+    } catch (err) {
+      // Connect to an error display component to show user that an error has occurred
+      console.log(err);
+    }
+  }
+
+  zipAttachment() {
```

Comment:
this function needs to be share by any components.I think this function needs be resusable . For future it can be use in Course